foxtrot-core/src/main/java/com/flipkart/foxtrot/core/table/impl/DistributedTableMetadataManager.java

```diff
@@ -268,7 +262,7 @@ private void estimateCardinality(final String table, final Collection<FieldMetad
                 case STRING: {
                     query.addAggregation(AggregationBuilders.cardinality(field)
                             .field(field)
-                            .precisionThreshold(5));
```

Comment:
The default internal precision is 14, which translates to ~3000 distinct values. This means that the default precision will count <3000 values with a hashmap and be exact, and anything over 3000 values will start approximating with HLL.

However, the minimum internal precision is 4, which translates to a cardinality of 2 before it starts approximating.

Value 5 seems to be too less

https://discuss.elastic.co/t/default-precision-of-cardinality-aggregation/51665/2

Reply:
Can we have these variables in some config file or db so that we can tune them later on without  sending a build